web/pgadmin/tools/sqleditor/static/js/components/sections/Query.jsx

```diff
@@ -508,9 +515,51 @@ export default function Query() {
     eventBus.fireEvent(QUERY_TOOL_EVENTS.QUERY_CHANGED, isDirty());
   }, []);
 
+  const checkViewEditDataPromotion = (args, event) => {
+
+    queryToolCtx.modal.showModal(gettext('Promote to Query Tool'), (closeModal) =>(
+      <ConfirmPromotionContent
+        closeModal={closeModal}
+        text={'Manually editing the query will cause this View/Edit Data tab to be converted to a Query Tool tab. You will be able to edit the query text freely, but no longer be able to use the toolbar buttons for sorting and filtering data. </br> Do you wish to continue?'}
+        onContinue={(formData)=>{
+          promoteToQueryTool();
+          // Use replaceSelection as user can add single character or select text and enter new values
+          editor.current.replaceSelection(event.key);
+          preferencesStore.setPreference(formData);
+          setIsPromoted((prev)=> {
```

Comment:
there is no need to maintain state for ispromoted. You can do the keydown unbind there itself.